Extension/src/LanguageServer/extension.ts

```diff
@@ -1413,13 +1413,33 @@ export async function preReleaseCheck(): Promise<void> {
     }
 }
 
-export async function getIncludes(maxDepth: number): Promise<any> {
-    const tokenSource = new vscode.CancellationTokenSource();
-    const includes = await clients.ActiveClient.getIncludes(maxDepth, tokenSource.token);
-    tokenSource.dispose();
+export async function getIncludesWithCancellation(maxDepth: number, token: vscode.CancellationToken): Promise<GetIncludesResult> {
+    const includes = await clients.ActiveClient.getIncludes(maxDepth, token);
+    const wksFolder = clients.ActiveClient.RootUri
+        ? vscode.workspace.workspaceFolders?.find(folder =>
+            clients.ActiveClient.RootUri?.toString().startsWith(folder.uri.toString())
+        )?.uri
+        : null;
+
+    if (!wksFolder) {
+        return includes;
+    }
+
+    const wksFolderString = wksFolder.toString();
+    includes.includedFiles = includes.includedFiles.filter(header => vscode.Uri.file(header).toString().startsWith(wksFolderString));
```

Comment:
I guess you're filtering out everything not under the 'primary' workspace folder?  That may filter out much more than standard headers.  It would seem to filter out all headers under other roots in a multiroot workspace.  And users may being using libraries, etc., that are not stored under their workspace folder.

Reply:
Thanks! I dropped the search for the current workspace, and assume that the RootUri is the one.

Reply:
I don't think it's a different approach. It's just fixing a bug with the original approach.

It should not be filtering out headers that are included from other user workspace folders.

Reply:
@Colengms  @sean-mcmanus we could discuss more offline about this since it would take time, but it is an interesting topic. In short, increasing dramatically the count of headers would not necessarily increase the accuracy of our algorithms (rather the opposite).
Hence while I totally agree with you that trying to exclude only the system headers is a good idea we should experiment in the future with. As of now, we only want to restore the functionality exactly as it was and as with got positive results from.